### PR TITLE
book details and card issues fixed

### PR DESCRIPTION
**Pull request by olgakahraman**

### CODE REVIEW

File: src/components/BookCard/BookCard.jsx [hunk 1]
```diff
@@ -12,8 +12,13 @@ export default function BookCard({
 			<img src={`../images/${coverImg}`} />
 			<div className="card-content">
 				<p className="card-title"> {title}</p>
-				<p className="card-author">
-					By {authors} ({publishDate})
+				<p className="book-author">
+					{Array.isArray(authors)
+						? `By ${authors.join(', ')}`
+						: `By ${authors || 'Unknown Author'}`}{' '}
```
Comment (ElijahMSmith) [inline src/components/BookCard/BookCard.jsx]: I think we can delete the extra space `{' '}` at the end of this line.

File: src/components/BookDetails/BookDetails.jsx [hunk 2]
```diff
@@ -15,10 +15,25 @@ export default function BookDetails({
 			<div className="book-details">
 				<p className="book-title">{title}</p>
 				<p className="book-author">
-					By {authors} ({publishDate})
+					{Array.isArray(authors)
+						? `By ${authors.join(', ')}`
+						: `By ${authors || 'Unknown Author'}`}{' '}
```
Comment (ElijahMSmith) [inline src/components/BookDetails/BookDetails.jsx]: Same here, we also don't need a space on the end.

File: src/components/BookDetails/BookDetails.jsx [hunk 3]
```diff
@@ -15,10 +15,25 @@ export default function BookDetails({
 			<div className="book-details">
 				<p className="book-title">{title}</p>
 				<p className="book-author">
-					By {authors} ({publishDate})
+					{Array.isArray(authors)
```
Comment (ElijahMSmith) [inline src/components/BookDetails/BookDetails.jsx]: Can we fix the model here to always make "authors" and "categories" an array of strings? We shouldn't have data that can be multiple different types. If there is only one author or category, it should be an array of length 1.

Might be worth double checking with Ievgeniia and Ivan to make sure this is the expected model.

File: src/components/BookDetails/BookDetails.jsx [hunk 4]
```diff
@@ -15,10 +15,25 @@ export default function BookDetails({
...
+						? `By ${authors.join(', ')}`
+						: `By ${authors || 'Unknown Author'}`}{' '}
+					{publishDate
+						? `(${publishDate})`
+						: '(No publish date)'}
 				</p>
 				<p className="book-genre">
-					<span>{categories}</span>
+					{Array.isArray(categories)
```
Comment (ElijahMSmith) [inline src/components/BookDetails/BookDetails.jsx]: See comment above about making this always an array. This is a really complicated statement just to render a list of strings. We should try to reduce the complexity here!

**Review by ElijahMSmith — APPROVED** [review]

File: src/components/BookCard/BookCard.jsx (hunk 1 above)
Comment (olgakahraman) [reply src/components/BookCard/BookCard.jsx]: i think they appear after format on save but i'll check it out

**Pull request merged**